src/axom/core/ArrayBase.hpp

```diff
@@ -728,20 +732,18 @@ struct ArrayOpsBase;
 template <typename T>
 struct ArrayOpsBase<T, false>
 {
-  static constexpr bool DefaultCtor = std::is_default_constructible<T>::value;
+  using DefaultCtorTag = std::is_default_constructible<T>;
 
   /*!
-   * \brief Default-initializes the "new" segment of an array
+   * \brief Helper for default-initializing the "new" segment of an array
    *
    * \param [inout] data The data to initialize
    * \param [in] begin The beginning of the subset of \a data that should be initialized
    * \param [in] end The end index (exclusive) of the subset of \a data that is to be initialized
```

Comment:
Please update the doxygen from `end` to `nelems`   (here and elsewhere).

Note: It looks like this was out of date before this PR, but it would still be good to fix it here.